### ISSUE TITLE
Bad Request on fastlane match 

### ISSUE THREAD

**Issue by paulsjohnson91** [fastlane/fastlane#17155]
May be related to [17116](https://github.com/fastlane/fastlane/issues/17116) however I'm not running `produce enable_services` first.

When running `fastlane match appstore`, after putting in credentials I'm getting

```
[14:58:09]: 'match AppStore <id>' is available on the Developer Portal, however it's 'Invalid', fixing this now for you 🔨
[14:58:10]: An error occurred while verifying your certificates and profiles with the Apple Developer Portal.
[14:58:10]: If you already have your certificates stored in git, you can run `fastlane match` in readonly mode
[14:58:10]: to just install the certificates and profiles without accessing the Dev Portal.
[14:58:10]: To do so, just pass `readonly: true` to your match call.
        24: from /usr/local/Cellar/fastlane/2.157.2/libexec/bin/fastlane:23:in `<main>'
        23: from /usr/local/Cellar/fastlane/2.157.2/libexec/bin/fastlane:23:in `load'
        22: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/bin/fastlane:23:in `<top (required)>'
        21: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/fastlane/lib/fastlane/cli_tools_distributor.rb:111:in `take_off'
        20: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/match/lib/match/commands_generator.rb:25:in `start'
        19: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/match/lib/match/commands_generator.rb:187:in `run'
        18: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
        17: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
        16: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
        15: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
        14: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
        13: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/match/lib/match/commands_generator.rb:68:in `block (3 levels) in run'
        12: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/match/lib/match/runner.rb:107:in `run'
        11: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/match/lib/match/runner.rb:107:in `each'
        10: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/match/lib/match/runner.rb:108:in `block in run'
         9: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/match/lib/match/runner.rb:108:in `loop'
         8: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/match/lib/match/runner.rb:109:in `block (2 levels) in run'
         7: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/match/lib/match/runner.rb:281:in `fetch_provisioning_profile'
         6: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/match/lib/match/spaceship_ensure.rb:85:in `profile_exists'
         5: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/spaceship/lib/spaceship/connect_api/models/profile.rb:87:in `delete!'
         4: from /usr/local/Cellar/ruby/2.7.1_2/lib/ruby/2.7.0/forwardable.rb:235:in `delete_profile'
         3: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/spaceship/lib/spaceship/connect_api/provisioning/provisioning.rb:95:in `delete_profile'
         2: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/spaceship/lib/spaceship/connect_api/provisioning/client.rb:45:in `delete'
         1: from /usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/spaceship/lib/spaceship/connect_api/provisioning/client.rb:87:in `proxy_delete'
/usr/local/Cellar/fastlane/2.157.2/libexec/gems/fastlane-2.157.2/spaceship/lib/spaceship/connect_api/api_client.rb:183:in `handle_response': Bad Request (Spaceship::UnexpectedResponse)
        24: from /usr/local/Cellar/fastlane/2.157.2/libexec/bin/fastlane:23:in `<main>'

```

My fastlane env:
```
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                   |
| --------------------------- | ------------------------------------------------------- |
| OS                          | 10.15.6                                                 |
| Ruby                        | 2.7.1                                                   |
| Bundler?                    | false                                                   |
| Git                         | git version 2.24.3 (Apple Git-128)                      |
| Installation Source         | /usr/local/Cellar/fastlane/2.157.2/libexec/bin/fastlane |
| Host                        | Mac OS X 10.15.6 (19G2021)                              |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.7.1_2/lib                      |
| OpenSSL Version             | OpenSSL 1.1.1f  31 Mar 2020                             |
| Is contained                | false                                                   |
| Is homebrew                 | false                                                   |
| Is installed via Fabric.app | false                                                   |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/             |
| Xcode Version               | 11.7                                                    |


### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |


### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
opt_out_usage

PROJECT_NAME = "<projectName>"
XCODE_PROJECT = "./ios/#{PROJECT_NAME}.xcodeproj"
XCODE_WORKSPACE = "./ios/#{PROJECT_NAME}.xcworkspace"
OUTPUT_DIRECTORY = "./fastlane/builds/"
BUILD_FOR_DEVELOPMENT_ONLY = false # true = Build for Development | false = Build to create a release
BUILD_ONLY = false
KEY_USERNAME = "<username>"
SCHEME_NAME = "<scheme name>"
BUMP_TYPE = ENV["BUMP_TYPE"]

platform :ios do


  private_lane :pushChangesBack do | options |
    if options[:running_on_ci]
      repository_uri = sh("git remote show origin | awk 'NR==2{print $3}' | sed 's/git@//g' | sed 's#https://##g'", log: false)
      push_uri = "https://user:#{options[:token]}@#{repository_uri.chomp}"
      branches = "HEAD:#{options[:git_branch_name]}"

      sh("git push #{push_uri} #{branches} --tags", log: false)
    else
      push_to_git_remote
    end

  end

  private_lane :certificates do | options |
    match_type = options[:match_type]

    if options[:ci]
      create_keychain(
        name: ENV["CI_KEYCHAIN_NAME"],
        password: ENV["CI_KEYCHAIN_PASSWORD"],
        default_keychain: true,
        unlock: true,
        timeout: 3600,
        lock_when_sleeps: false
      )
      match(
        type: match_type,
        keychain_name: ENV["CI_KEYCHAIN_NAME"],
        keychain_password: ENV["CI_KEYCHAIN_PASSWORD"],
        readonly: true,
        clone_branch_directly: true,
        shallow_clone: true,
        verbose: false,
      )
    else
      match(
        type: match_type,
        readonly: true,
        shallow_clone: true,
        verbose: false,
        clone_branch_directly: true,
        git_url: "<url<",
        app_identifier: "<id>"
      )
    end

  end

  private_lane :build do | options |
    build_for_development_only = options[:build_for_development_only]
    configuration_name = build_for_development_only ? "Development" : "Release"
    export_options_method = build_for_development_only ? "development" : "app-store"
    match_type = build_for_development_only ? "development" : "appstore"

    certificates(
      ci: options[:running_on_ci],
      match_type: match_type,
    )

    cocoapods(
      repo_update: true,
      podfile: "./ios/Podfile"
    )
    gym(
      scheme: SCHEME_NAME,
      workspace: XCODE_WORKSPACE,
      silent: true,
      clean: true,
      configuration: configuration_name,
      output_directory: OUTPUT_DIRECTORY,
      output_name: "#{PROJECT_NAME}.ipa",
    )

    match_type
  end

  private_lane :ship do | options |
    pilot(
      ipa: "#{OUTPUT_DIRECTORY}#{PROJECT_NAME}.ipa",
      skip_waiting_for_build_processing: true,
      verbose: false,
      uses_non_exempt_encryption: true,
      notify_external_testers: false,
      username: options[:apple_id],
      team_id: options[:apple_team_id],
      apple_id: options[:apple_app_id],
      itc_provider: ENV["sigh_#{options[:app_identifier]}_#{options[:match_type]}_team-id"],
    )
  end

  lane :updateProfile
     update_project_provisioning(
        xcodeproj: XCODE_PROJECT,
        build_configuration: "Release"
     )
  end

  lane :commitTest do
    running_on_ci = true
    options = {
      :build_for_development_only => BUILD_FOR_DEVELOPMENT_ONLY,
      :running_on_ci => running_on_ci,
      :git_branch_name => running_on_ci ? "#{ENV['GIT_BRANCH_NAME']}"  : sh("git", "rev-parse", "--abbrev-ref", "HEAD", log: false),
      :app_identifier => ENV["APP_IDENTIFIER"],
      :apple_id => ENV["APPLE_ID"],
      :apple_team_id => ENV["APPLE_TEAM_ID"],
      :apple_app_id => ENV["APPLE_APP_ID"],
      :token => ENV["GITHUB_TOKEN"],
    }
    increment_build_number(
      xcodeproj: XCODE_PROJECT,
    )
    pushChangesBack(options)
  end

  lane :buildAndShip do
    running_on_ci = ENV["CI"] || false
    options = {
      :build_for_development_only => BUILD_FOR_DEVELOPMENT_ONLY,
      :running_on_ci => running_on_ci,
      :git_branch_name => running_on_ci ? "#{ENV['GIT_BRANCH_NAME']}"  : sh("git", "rev-parse", "--abbrev-ref", "HEAD", log: false),
      :app_identifier => ENV["APP_IDENTIFIER"],
      :apple_id => ENV["APPLE_ID"],
      :apple_team_id => ENV["APPLE_TEAM_ID"],
      :apple_app_id => ENV["APPLE_APP_ID"],
      :token => ENV["GITHUB_TOKEN"],
    }


    match_type = build(options)


    unless BUILD_ONLY
      ship(
        apple_id: options[:apple_id],
        apple_team_id: options[:apple_team_id],
        apple_app_id: options[:apple_app_id],
        match_type: match_type,
      )
    end
  end
end

```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.157.2 | 🚫 Update available |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.4.0        |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.3        |
| CFPropertyList            | 3.0.2        |
| claide                    | 1.0.3        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.3.0        |
| xcodeproj                 | 1.18.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| public_suffix             | 4.0.5        |
| addressable               | 2.7.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.8.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| babosa                    | 1.0.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.76.0       |
| faraday                   | 1.0.1        |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 1.0.0        |
| fastimage                 | 2.2.0        |
| gh_inspector              | 1.1.3        |
| json                      | 2.3.0        |
| mini_magick               | 4.10.1       |
| rubyzip                   | 2.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.7.6        |
| bundler                   | 2.1.2        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| jwt                       | 2.2.2        |
| uber                      | 0.1.0        |
| declarative               | 0.0.20       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| multi_json                | 1.15.0       |
| signet                    | 0.14.0       |
| memoist                   | 0.16.2       |
| os                        | 1.1.1        |
| googleauth                | 0.13.1       |
| httpclient                | 2.8.3        |
| google-api-client         | 0.38.0       |
| google-cloud-env          | 1.3.3        |
| google-cloud-errors       | 1.0.1        |
| google-cloud-core         | 1.5.0        |
| rake                      | 13.0.1       |
| digest-crc                | 0.6.1        |
| google-cloud-storage      | 1.28.0       |
| emoji_regex               | 3.0.0        |
| jmespath                  | 1.4.0        |
| aws-partitions            | 1.363.0      |
| aws-eventstream           | 1.1.0        |
| aws-sigv4                 | 1.2.2        |
| aws-sdk-core              | 3.105.0      |
| aws-sdk-kms               | 1.37.0       |
| aws-sdk-s3                | 1.79.1       |
| uri                       | 0.10.0       |
| forwardable               | 1.3.1        |
| logger                    | 1.4.2        |
| cgi                       | 0.1.0        |
| timeout                   | 0.1.0        |
| stringio                  | 0.1.0        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.2.0        |
| strscan                   | 1.0.3        |
| date                      | 3.0.0        |
| delegate                  | 0.1.0        |
| fileutils                 | 1.4.1        |
| io-console                | 0.5.6        |
| zlib                      | 1.1.0        |
| singleton                 | 0.1.0        |
| rexml                     | 3.2.3        |
| open3                     | 0.1.0        |
| yaml                      | 0.1.0        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
| webrick                   | 1.6.0        |
</details>

*generated on:* **2020-09-04**
</details>
```

**Comment by joshdholtz** [comment]
Looking into right away. Sorry fam 😔

**Comment by joshdholtz** [comment]
This should be fixed in `2.157.3` but might also want to wait until #17160 gets merged 😇

**Comment by paulsjohnson91** [comment]
Thanks for the quick turnaround!
